usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -78,7 +78,7 @@ func processScenarioIteration(ctx context.Context, params ScenarioEvaluationPara
 		dataAccessor.ClientObject,
 		params.DataModel,
 	)
-	if errEval != nil && !errors.Is(errEval, models.ErrScenarioTriggerConditionAndTriggerObjectMismatch) {
```

Comment:
regression fix: part of the function computing the scenario execution scenario was factorized, but this is not where the check on the error type should be done. As a result, scenarios were executed even if the trigger condition does not pass.